cms/server/templates/contest/task_submissions.html

```diff
@@ -101,21 +101,41 @@ <h2 style="margin-bottom: 10px">{{ _("Submit a solution") }}</h2>
 {% end %}
 
 <div id="submit_solution" class="row">
-    <div class="span5">
+    <div class="span9">
         <form class="form-horizontal" enctype="multipart/form-data" action="{{ url_root }}/tasks/{{ encode_for_url(task.name) }}/submit" method="POST">
-            <fieldset>
+            <fieldset class="row-fluid">
 {% for idx, filename in enumerate(x.filename for x in task.submission_format) %}
                 <div class="control-group">
-                    <label class="control-label" for="input{{ idx }}">{{ filename.replace(".%l", "") }}</label>
-                    <div class="controls">
+                        <label class="control-label span2" for="input{{ idx }}">{{ filename.replace(".%l", "") }}</label>
+
+                    <div class="span4">
                         <input type="file" class="input-xlarge" id="input{{ idx }}" name="{{ filename }}"/>
+                        <!-- FIXME: figure out why we had div class="controls" wrapping this -->
                     </div>
+                    <!-- Kek -->
+
+                    <!-- FIXME: Do not show this if task type is output only -->
```

Comment:
In theory this page should have access to 'task' so you can probably do something dodgy with `isinstance` or `task.__class__.__name__`